src/Generators/Microsoft.Gen.Logging/Parsing/TemplateProcessor.cs

```diff
@@ -86,56 +129,68 @@ internal static void ExtractTemplates(string? message, List<string> templates)
     internal static int FindIndexOfAny(string message, char[] chars, int startIndex, int endIndex)
     {
         var findIndex = message.IndexOfAny(chars, startIndex, endIndex - startIndex);
-        return findIndex == -1
-            ? endIndex
-            : findIndex;
+        return findIndex == -1 ? endIndex : findIndex;
     }
 
-    private static int FindBraceIndex(string message, char brace, int startIndex, int endIndex)
+    /// <summary>
+    /// Searches for the next brace index in the message.
+    /// </summary>
+    /// <remarks> The search skips any sequences of {{ or }}.</remarks>
+    /// <example>{{prefix{{{Argument}}}suffix}}.</example>
+    /// <returns>The zero-based index position of the first occurrence of the searched brace; -1 if the searched brace was not found; -2 if the wrong brace was found.</returns>
```

Comment:
Perhaps, it makes sense to introduce consts for the "magic" numbers. It'd make the code a little easier to comprehend.

Reply:
I get it for the part of returning the index -1, or -2. But it also complains about `bracesCount % 2` 😄 
Is that really a magic number, it's not like it would be anything else if checking for evens.

Reply:
I see a value of introducing consts for -1 and -2 as those are used in few places (including the docs). It'll make the maintenance easier.
`bracesCount % 2` can/should be supressed, as you do.